colfetcher: incomplete handling of not needed vectors

sql/tests.TestRandomSyntaxSQLSmith [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2949835&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=2949835&tab=artifacts#/) on master @ [f9dca40d3afb0a605eeee76bec4a07ff65d6fec0](https://github.com/cockroachdb/cockroach/commits/f9dca40d3afb0a605eeee76bec4a07ff65d6fec0):


```
        SET CLUSTER SETTING sql.defaults.drop_enum_value.enabled = true;
        SET enable_drop_enum_value = true;
        
        		SET CLUSTER SETTING sql.stats.automatic_collection.enabled = false;
        		SET CLUSTER SETTING sql.stats.histogram_collection.enabled = false;
        		SET CLUSTER SETTING sql.defaults.interleaved_tables.enabled = true;
        	CREATE TABLE table1 (col1_0 FLOAT4 NOT NULL, PRIMARY KEY (col1_0 DESC), INDEX (col1_0 ASC) WHERE table1.col1_0 != 0.0:::FLOAT8, UNIQUE (col1_0 ASC) WHERE table1.col1_0 = 0.0:::FLOAT8, UNIQUE (col1_0), UNIQUE (col1_0) WHERE table1.col1_0 >= 1.0:::FLOAT8, UNIQUE (col1_0 ASC), INDEX (col1_0 DESC), INDEX (col1_0) WHERE table1.col1_0 < 1.0:::FLOAT8);
        CREATE TABLE table2 (col2_0 REGNAMESPACE NOT NULL, col2_1 VARCHAR NOT NULL, col2_2 INET NOT NULL, col2_3 CHAR NOT NULL, col2_4 INTERVAL NOT NULL, col2_5 BIT(20) NOT NULL, col2_6 VARCHAR NOT NULL, col2_7 INET NOT NULL, col2_8 "char" NOT NULL, col2_9 REGTYPE NOT NULL, PRIMARY KEY (col2_3, col2_7 DESC, col2_4 DESC, col2_9 ASC, col2_0 DESC, col2_2 ASC, col2_6 DESC, col2_8 ASC, col2_5, col2_1 ASC), col2_10 STRING NOT NULL AS (lower(col2_8)) STORED, col2_11 STRING NOT NULL AS (lower(col2_1)) STORED, col2_12 STRING NOT NULL AS (lower(col2_1)) STORED, col2_13 STRING NOT NULL AS (lower(CAST(col2_5 AS STRING))) VIRTUAL, UNIQUE (col2_10 DESC), UNIQUE (col2_1, col2_4, col2_5 ASC, col2_3 DESC, col2_11 DESC, col2_12 ASC, col2_8 ASC, col2_13 ASC, col2_9 DESC, col2_0, col2_7 ASC, col2_10 DESC), INDEX (col2_0 ASC, col2_4, col2_1, col2_12 ASC, col2_10 ASC, col2_8 DESC, col2_9, col2_13, col2_2, col2_11) WHERE ((((table2.col2_10 != 'X':::STRING) AND (table2.col2_8 >= 'X':::STRING)) AND (table2.col2_6 >= '':::STRING)) OR (table2.col2_3 != e'\U00002603':::STRING)) AND (table2.col2_13 = e'\U00002603':::STRING), INDEX (col2_12 ASC, col2_0 DESC, col2_7 ASC, col2_8 ASC, col2_3, col2_6, col2_2 DESC, col2_11 ASC, col2_10 DESC, col2_4 ASC, col2_9 DESC, col2_13 ASC, col2_5 ASC, col2_1) WHERE ((((((table2.col2_10 <= 'X':::STRING) OR (table2.col2_11 <= e'\x00':::STRING)) AND (table2.col2_6 = '':::STRING)) OR (table2.col2_1 >= '':::STRING)) AND (table2.col2_13 <= e'\'':::STRING)) AND (table2.col2_12 = e'\'':::STRING)) AND (table2.col2_3 = e'\U00002603':::STRING), UNIQUE (col2_11, col2_1 ASC, col2_4 ASC, col2_8, col2_7 ASC, col2_3) WHERE ((((((table2.col2_13 > '':::STRING) AND (table2.col2_1 != e'\'':::STRING)) OR (table2.col2_6 = e'\U00002603':::STRING)) OR (table2.col2_8 < '"':::STRING)) AND (table2.col2_10 >= e'\U00002603':::STRING)) AND (table2.col2_3 >= e'\'':::STRING)) AND (table2.col2_12 < '"':::STRING), INDEX (col2_13, col2_6 ASC, col2_11 DESC, col2_4 DESC, col2_1 ASC, col2_7 ASC, col2_8 ASC, col2_2, col2_9 DESC, col2_10, col2_12 DESC, col2_5, col2_0) WHERE (((((((table2.col2_10 != '"':::STRING) OR (table2.col2_1 <= 'X':::STRING)) AND (table2.col2_13 != e'\U00002603':::STRING)) AND (table2.col2_12 > e'\'':::STRING)) AND (table2.col2_6 > e'\'':::STRING)) AND (table2.col2_8 < e'\'':::STRING)) AND (table2.col2_11 >= 'X':::STRING)) OR (table2.col2_3 != 'X':::STRING), INDEX (col2_2 DESC, col2_7 ASC, col2_4, col2_5, col2_11 DESC, col2_10 ASC, col2_6 ASC, col2_9, col2_3 ASC, col2_12 DESC, col2_0 ASC), INDEX (col2_3, col2_9 DESC, col2_7 ASC) STORING (col2_10) WHERE (((table2.col2_6 = 'X':::STRING) AND (table2.col2_13 < '"':::STRING)) OR (table2.col2_1 < 'X':::STRING)) OR (table2.col2_10 < '':::STRING), FAMILY (col2_10, col2_2), FAMILY (col2_12), FAMILY (col2_6), FAMILY (col2_7), FAMILY (col2_0), FAMILY (col2_5, col2_9, col2_3), FAMILY (col2_1), FAMILY (col2_11, col2_8), FAMILY (col2_4));
        ALTER TABLE table1 INJECT STATISTICS '[{"columns": ["col1_0"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 164228, "histo_col_type": "FLOAT4", "name": "__auto__", "null_count": 0, "row_count": 700000}]':::JSONB;
        ALTER TABLE table2 INJECT STATISTICS e'[{"columns": ["col2_8"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1663, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 3000}, {"columns": ["col2_4"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2401, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 3000}, {"columns": ["col2_6"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2798, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 3000}, {"columns": ["col2_10"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2289, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 3000}, {"columns": ["col2_11"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 192, "histo_buckets": [{"distinct_range": 0, "num_eq": 3107066348745136756, "num_range": 0, "upper_bound": "\\u0000"}], "histo_col_type": "STRING", "name": "__auto__", "null_count": 0, "row_count": 3000}, {"columns": ["col2_13"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1056, "histo_buckets": [{"distinct_range": 0, "num_eq": 7240327964891365451, "num_range": 0, "upper_bound": "\\u0014\\u000ef\'D\\u001bRj4"}, {"distinct_range": 711399252880235300, "num_eq": 8327784297973340280, "num_range": 711399252880235307, "upper_bound": "srO\\u0002d\\u0014d\\u0013>"}], "histo_col_type": "STRING", "name": "__auto__", "null_count": 0, "row_count": 3000}, {"columns": ["col2_2"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1582, "histo_buckets": [{"distinct_range": 0, "num_eq": 3831033851152455878, "num_range": 0, "upper_bound": "118.251.203.32/5"}, {"distinct_range": 8564800485481229000, "num_eq": 80000000, "num_range": 8564800485481229213, "upper_bound": "135.193.191.21/10"}, {"distinct_range": 2012199871076234800, "num_eq": 5000, "num_range": 4924963374547161391, "upper_bound": "c3db:7d0f:b160:6b46:b4f4:a1b:a8a9:7417/111"}], "histo_col_type": "INET", "name": "__auto__", "null_count": 0, "row_count": 3000}, {"columns": ["col2_3"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 431, "histo_buckets": [{"distinct_range": 0, "num_eq": 8650552567708441013, "num_range": 0, "upper_bound": "\\u0011"}, {"distinct_range": 7320115737005092000, "num_eq": 800, "num_range": 7615880907883749025, "upper_bound": "\\u0015"}, {"distinct_range": 5245.105949820407, "num_eq": 5823866732610089098, "num_range": 6000, "upper_bound": "\\u0019"}, {"distinct_range": 0, "num_eq": 1035202996120692621, "num_range": 0, "upper_bound": "!"}, {"distinct_range": 0, "num_eq": 500000, "num_range": 1642434645332730790, "upper_bound": "$"}, {"distinct_range": 0, "num_eq": 30000000, "num_range": 3216112055474359157, "upper_bound": "\'"}, {"distinct_range": 0, "num_eq": 3000000000, "num_range": 500000000, "upper_bound": "2"}, {"distinct_range": 1853134305446131700, "num_eq": 7672953661136189747, "num_range": 1894095716249544017, "upper_bound": "J"}, {"distinct_range": 1998331714736339200, "num_eq": 2121915506116748154, "num_range": 5808250913920456689, "upper_bound": "y"}], "histo_col_type": "CHAR", "name": "__auto__", "null_count": 0, "row_count": 3000}, {"columns": ["col2_5"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 895, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 3000}, {"columns": ["col2_12"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1398, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 3000}, {"columns": ["col2_0"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 136, "histo_buckets": [{"distinct_range": 0, "num_eq": 8000, "num_range": 0, "upper_bound": "0"}, {"distinct_range": 8000000000, "num_eq": 40, "num_range": 8000000000, "upper_bound": "1002883938"}, {"distinct_range": 0, "num_eq": 6474396310918104859, "num_range": 0, "upper_bound": "1750183413"}, {"distinct_range": 2644955897.7562723, "num_eq": 40000000000, "num_range": 3000000000, "upper_bound": "1926876690"}, {"distinct_range": 0, "num_eq": 700000000, "num_range": 0, "upper_bound": "2981056255"}, {"distinct_range": 77.54695945430613, "num_eq": 700000, "num_range": 300, "upper_bound": "3623118858"}, {"distinct_range": 800000000, "num_eq": 30, "num_range": 800000000, "upper_bound": "3963993546"}], "histo_col_type": "REGNAMESPACE", "name": "__auto__", "null_count": 0, "row_count": 3000}, {"columns": ["col2_1"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 698, "histo_buckets": [{"distinct_range": 0, "num_eq": 6990581367730671162, "num_range": 0, "upper_bound": ""}, {"distinct_range": 257199763841910530, "num_eq": 9120063886631880338, "num_range": 4172902844091929486, "upper_bound": "\\u001bUuB"}, {"distinct_range": 80000000, "num_eq": 60000000, "num_range": 80000000, "upper_bound": "4Y/\\u001f*Q i"}, {"distinct_range": 60000000000, "num_eq": 1987959061205168809, "num_range": 60000000000, "upper_bound": "@Vc>\\u0017"}, {"distinct_range": 3599252179557993000, "num_eq": 7000000, "num_range": 6729234864688546942, "upper_bound": "JyZ\';A,{"}, {"distinct_range": 5000000, "num_eq": 70000000000, "num_range": 5000000, "upper_bound": "ot"}, {"distinct_range": 90062849715.2711, "num_eq": 900000000, "num_range": 100000000000, "upper_bound": "wi^t"}, {"distinct_range": 0, "num_eq": 856382530121843520, "num_range": 2057481219463626862, "upper_bound": "}"}], "histo_col_type": "VARCHAR", "name": "__auto__", "null_count": 0, "row_count": 3000}, {"columns": ["col2_7"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1172, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 3000}, {"columns": ["col2_9"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2303, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 3000}]':::JSONB;
        CREATE TYPE rand_typ_0 AS ENUM ('lex');
        CREATE TYPE rand_typ_1 AS ENUM ('woxm');
        CREATE TYPE rand_typ_2 AS ENUM ('b', 'hzapqz', 'q', 'kxu', 'rnrxxa', 'zrho');
        CREATE TYPE rand_typ_3 AS ENUM ('sx');
        ;
    rsg_test.go:577: 
        SET CLUSTER SETTING sql.defaults.drop_enum_value.enabled = true;
        SET enable_drop_enum_value = true;
        
        CREATE TYPE greeting AS ENUM ('hello', 'howdy', 'hi', 'good day', 'morning');
        CREATE TABLE IF NOT EXISTS seed AS
        	SELECT
        		g::INT2 AS _int2,
        		g::INT4 AS _int4,
        		g::INT8 AS _int8,
        		g::FLOAT4 AS _float4,
        		g::FLOAT8 AS _float8,
        		'2001-01-01'::DATE + g AS _date,
        		'2001-01-01'::TIMESTAMP + g * '1 day'::INTERVAL AS _timestamp,
        		'2001-01-01'::TIMESTAMPTZ + g * '1 day'::INTERVAL AS _timestamptz,
        		g * '1 day'::INTERVAL AS _interval,
        		g % 2 = 1 AS _bool,
        		g::DECIMAL AS _decimal,
        		g::STRING AS _string,
        		g::STRING::BYTES AS _bytes,
        		substring('00000000-0000-0000-0000-' || g::STRING || '00000000000', 1, 36)::UUID AS _uuid,
        		'0.0.0.0'::INET + g AS _inet,
        		g::STRING::JSONB AS _jsonb,
        		enum_range('hello'::greeting)[g] as _enum
        	FROM
        		generate_series(1, 5) AS g;
        
        INSERT INTO seed DEFAULT VALUES;
        CREATE INDEX on seed (_int8, _float8, _date);
        CREATE INVERTED INDEX on seed (_jsonb);
        ;
    rsg_test.go:579: 
    rsg_test.go:580: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestRandomSyntaxSQLSmith917147790
--- FAIL: TestRandomSyntaxSQLSmith (300.60s)
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
make stressrace TESTS=TestRandomSyntaxSQLSmith PKG=./pkg/sql/tests TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #64592 sql/tests: TestRandomSyntaxSQLSmith failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<details><summary>Internal log</summary>
<p>

```
mjibson marked as alumn{us/a}; resolving to rafiss instead
```
</p>
</details>

/cc @cockroachdb/sql-queries @rafiss
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxSQLSmith.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/tests.TestRandomSyntaxSQLSmith [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2954091&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=2954091&tab=artifacts#/) on master @ [43a079db6dc1e6460a0bdb23a73bb69c81389532](https://github.com/cockroachdb/cockroach/commits/43a079db6dc1e6460a0bdb23a73bb69c81389532):

Random syntax error:

```
    rsg_test.go:758: Crash detected: server panic: pq: internal error: unexpected error from the vectorized engine: uuid: UUID must be exactly 16 bytes long, got 0 bytes
```
Query:

```
        SELECT
        	true AS col_6286,
        	tab_2720._timestamp AS col_6287,
        	'"':::STRING AS col_6288,
        	tab_2720._int2 AS col_6289,
        	(-0.3179420053293992417):::DECIMAL AS col_6290,
        	NULL AS col_6291,
        	'hello':::greeting AS col_6292,
        	230108557:::INT8 AS col_6293,
        	tab_2720._int8 AS col_6294,
        	tab_2720._interval AS col_6295,
        	'9614c256-98f7-44fc-806f-8e19d4454b0a':::UUID AS col_6296,
        	NULL AS col_6297,
        	(tab_2720._date::DATE + tab_2720._int8::INT8)::DATE AS col_6298
        FROM
        	defaultdb.public.seed@[0] AS tab_2720;
```
Schema:

```
    rsg_test.go:575: To reproduce, use schema:
    rsg_test.go:577: 
        SET CLUSTER SETTING sql.defaults.drop_enum_value.enabled = true;
        SET enable_drop_enum_value = true;
        
        		SET CLUSTER SETTING sql.stats.automatic_collection.enabled = false;
        		SET CLUSTER SETTING sql.stats.histogram_collection.enabled = false;
        		SET CLUSTER SETTING sql.defaults.interleaved_tables.enabled = true;
        	CREATE TABLE table1 (col1_0 FLOAT4 NULL, col1_1 UUID NOT NULL, col1_2 INET, col1_3 REGNAMESPACE NOT NULL, col1_4 TIMETZ[] NULL, col1_5 REGPROC NOT NULL, col1_6 TIMESTAMP NULL, col1_7 FLOAT4 NOT NULL, col1_8 JSONB, col1_9 BOX2D NOT NULL, PRIMARY KEY (col1_1 ASC), col1_10 FLOAT4 NULL AS (col1_7 + col1_0) STORED, col1_11 FLOAT4 NULL AS (col1_0 + col1_7) VIRTUAL, col1_12 FLOAT4 NULL AS (col1_7 + col1_0) STORED, col1_13 STRING NULL AS (lower(CAST(col1_6 AS STRING))) VIRTUAL, col1_14 FLOAT4 NULL AS (col1_7 + col1_0) VIRTUAL, col1_15 FLOAT4 NULL AS (col1_0 + col1_7) STORED, col1_16 STRING NOT NULL AS (lower(CAST(col1_1 AS STRING))) STORED, col1_17 FLOAT4 NULL AS (col1_0 + col1_7) VIRTUAL, col1_18 STRING NOT NULL AS (lower(CAST(col1_9 AS STRING))) VIRTUAL, INDEX (col1_16) STORING (col1_2, col1_6, col1_8, col1_9, col1_10, col1_15) WHERE (((((((((((table1.col1_14 >= 3.4028234663852886e+38:::FLOAT8) OR (table1.col1_0 < 3.4028234663852886e+38:::FLOAT8)) AND (table1.col1_17 <= 0.0:::FLOAT8)) AND (table1.col1_18 <= 'X':::STRING)) AND (table1.col1_7 < '+Inf':::FLOAT8)) OR (table1.col1_16 < e'\'':::STRING)) OR (table1.col1_11 > 3.4028234663852886e+38:::FLOAT8)) OR (table1.col1_15 < 1.401298464324817e-45:::FLOAT8)) AND (table1.col1_6 > '-4713-11-24 00:00:00':::TIMESTAMP)) OR (table1.col1_12 = '-Inf':::FLOAT8)) AND (table1.col1_13 >= e'\'':::STRING)) AND (table1.col1_10 > 1.401298464324817e-45:::FLOAT8), INVERTED INDEX (col1_7 DESC, col1_9 DESC, col1_5, col1_10, col1_17, col1_3 ASC, col1_1 DESC, col1_12, col1_13 DESC, col1_16, col1_4 ASC) WHERE ((((table1.col1_12 = 0.0:::FLOAT8) AND (table1.col1_18 > 'X':::STRING)) OR (table1.col1_0 = 3.4028234663852886e+38:::FLOAT8)) AND (table1.col1_6 >= '0001-01-01 00:00:00':::TIMESTAMP)) AND (table1.col1_14 > 3.4028234663852886e+38:::FLOAT8), UNIQUE (col1_13 ASC, col1_12 ASC, col1_18, col1_0 ASC, col1_14 ASC, col1_2 ASC) STORING (col1_3, col1_4, col1_6, col1_7, col1_9, col1_15, col1_16) WHERE (((((((((((table1.col1_10 < 1.401298464324817e-45:::FLOAT8) OR (table1.col1_16 >= '"':::STRING)) OR (table1.col1_11 >= 3.4028234663852886e+38:::FLOAT8)) AND (table1.col1_17 > 0.0:::FLOAT8)) AND (table1.col1_18 > e'\'':::STRING)) OR (table1.col1_15 > 3.4028234663852886e+38:::FLOAT8)) OR (table1.col1_7 != 1.0:::FLOAT8)) OR (table1.col1_13 < e'\'':::STRING)) OR (table1.col1_14 = '+Inf':::FLOAT8)) AND (table1.col1_0 > '+Inf':::FLOAT8)) AND (table1.col1_12 <= 0.0:::FLOAT8)) OR (table1.col1_6 >= '-4713-11-24 00:00:00':::TIMESTAMP), FAMILY (col1_10), FAMILY (col1_0), FAMILY (col1_1), FAMILY (col1_4, col1_9), FAMILY (col1_5), FAMILY (col1_2), FAMILY (col1_7), FAMILY (col1_16, col1_8), FAMILY (col1_3), FAMILY (col1_12, col1_6, col1_15), UNIQUE (col1_16));
        CREATE TABLE table2 (col1_1 UUID NOT NULL, col2_1 REGTYPE NOT NULL, col2_2 INT2, col2_3 TIMESTAMP NOT NULL, col2_4 DATE NOT NULL, col2_5 "char" NULL, col2_6 BOOL, col2_7 BIT(37), col2_8 VARCHAR, col2_9 STRING NOT NULL, col2_10 FLOAT4, col2_11 CHAR NOT NULL, col2_12 BIT(24) NULL, col2_13 FLOAT8 NULL, col2_14 TIME NOT NULL, col2_15 INT4 NOT NULL, col2_16 "char", col2_17 TIMESTAMPTZ, PRIMARY KEY (col1_1 ASC, col2_12 DESC), UNIQUE (col2_6 DESC, col1_1 ASC, col2_1 DESC, col2_9, col2_8 DESC, col2_7, col2_11 ASC, col2_15 ASC, col2_12) WHERE (((((((((((table2.col2_5 != '':::STRING) AND (table2.col2_11 != '':::STRING)) AND (table2.col2_3 < '-4713-11-24 00:00:00':::TIMESTAMP)) OR (table2.col2_8 = '"':::STRING)) AND (table2.col2_9 <= e'\'':::STRING)) OR (table2.col2_4 >= '-infinity':::DATE)) AND (table2.col2_14 <= '00:00:00':::TIME)) OR (table2.col2_13 < 1.401298464324817e-45:::FLOAT8)) OR (table2.col2_17 <= '-4713-11-24 00:00:00+00:00':::TIMESTAMPTZ)) OR (table2.col2_10 <= '+Inf':::FLOAT8)) AND (NOT table2.col2_6)) AND (table2.col2_2 < (-32768):::INT8), UNIQUE (col2_1 DESC, col2_12 ASC, col2_16, col2_14, col2_7 ASC, col2_2 DESC, col2_4, col2_15, col2_13 ASC) WHERE table2.col2_4 <= '4714-11-24 BC':::DATE, UNIQUE (col2_13 ASC, col2_15 DESC, col2_16) STORING (col2_3, col2_4, col2_5, col2_7, col2_8, col2_9, col2_10, col2_14) WHERE (((((((table2.col2_15 = (-128):::INT8) OR (table2.col2_3 >= '-2000-01-01 00:00:00':::TIMESTAMP)) AND (table2.col2_13 > 1.401298464324817e-45:::FLOAT8)) AND (table2.col2_2 != 0:::INT8)) AND (table2.col2_10 <= 1.0:::FLOAT8)) OR (table2.col2_11 != e'\x00':::STRING)) OR (table2.col2_5 > e'\x00':::STRING)) OR table2.col2_6, UNIQUE (col2_1, col2_9 DESC, col2_16 ASC, col2_15 DESC, col2_8) STORING (col2_2, col2_4, col2_5, col2_7, col2_10, col2_11, col2_14) WHERE (table2.col2_4 = '4714-11-24 BC':::DATE) AND (table2.col2_9 != '"':::STRING), INDEX (col2_12 DESC, col2_1, col2_2 DESC, col2_6, col2_5, col2_7 DESC, col2_17 ASC, col2_16, col2_4, col2_13 ASC, col2_9, col2_14 DESC, col1_1 DESC) WHERE ((((((((((table2.col2_4 > '1970-01-01':::DATE) OR (table2.col2_14 != '24:00:00':::TIME)) AND (table2.col2_11 = 'X':::STRING)) OR (table2.col2_13 <= 3.4028234663852886e+38:::FLOAT8)) OR (table2.col2_17 < '0001-01-01 00:00:00+00:00':::TIMESTAMPTZ)) AND (table2.col2_10 <= 0.0:::FLOAT8)) OR (table2.col2_9 > e'\U00002603':::STRING)) AND (table2.col2_5 = e'\'':::STRING)) AND (table2.col2_16 != 'X':::STRING)) AND (table2.col2_15 <= 127:::INT8)) AND (NOT table2.col2_6), INDEX (col2_17, col2_7 DESC, col2_3, col2_10, col2_16, col2_12 DESC, col2_6 DESC, col2_2 DESC, col2_9 ASC, col2_14 ASC, col2_15 ASC, col2_1 ASC, col2_13, col1_1 ASC) WHERE (((((table2.col2_2 != 1:::INT8) AND (table2.col2_13 = 0.0:::FLOAT8)) OR (table2.col2_8 > '"':::STRING)) OR (table2.col2_16 <= e'\x00':::STRING)) OR (table2.col2_11 <= 'X':::STRING)) AND (NOT table2.col2_6), INDEX (col2_15, col2_11 DESC, col2_9) STORING (col2_1, col2_3, col2_5, col2_6, col2_7, col2_8, col2_16) WHERE ((((((((((table2.col2_4 != 'infinity':::DATE) OR (table2.col2_10 = '-Inf':::FLOAT8)) OR (table2.col2_9 <= e'\U00002603':::STRING)) OR (table2.col2_2 >= (-1):::INT8)) AND (table2.col2_11 > e'\x00':::STRING)) OR (table2.col2_17 < '-2000-01-01 00:00:00+00:00':::TIMESTAMPTZ)) AND (table2.col2_15 <= 1:::INT8)) OR (table2.col2_13 != 1.0:::FLOAT8)) OR (NOT table2.col2_6)) AND (table2.col2_5 <= e'\x00':::STRING)) AND (table2.col2_16 > '':::STRING), INDEX (col2_15 ASC, col2_13, col2_17, col2_1 DESC, col2_6, col2_5, col2_10 DESC, col2_11), FAMILY (col2_13), FAMILY (col2_12, col2_2, col2_17), FAMILY (col2_7), FAMILY (col2_3, col2_15), FAMILY (col1_1, col2_16, col2_1), FAMILY (col2_14, col2_11), FAMILY (col2_10, col2_5), FAMILY (col2_4), FAMILY (col2_6), FAMILY (col2_9, col2_8)) INTERLEAVE IN PARENT table1 (col1_1);
        CREATE TABLE table3 (col1_1 UUID NOT NULL, col3_1 VARCHAR NULL, col3_2 INT8 NOT NULL, col3_3 REGPROC NOT NULL, col3_4 BOX2D NULL, col3_5 INT8 NULL, col3_6 BOX2D NULL, col3_7 REGTYPE NOT NULL, col3_8 INET, col3_9 "char", col3_10 BIT NOT NULL, col3_11 REGTYPE NULL, col3_12 BOOL, col3_13 VARCHAR, col3_14 FLOAT8 NULL, col3_15 INTERVAL NOT NULL, col3_16 FLOAT4 NOT NULL, PRIMARY KEY (col1_1 ASC, col3_12 DESC, col3_7 ASC), UNIQUE (col3_7, col1_1 ASC, col3_8 ASC, col3_4, col3_1 DESC, col3_10 DESC, col3_3 ASC, col3_14, col3_13 ASC, col3_6 DESC, col3_5 ASC, col3_11 ASC) STORING (col3_2), INDEX (col3_5 DESC, col3_4 DESC, col1_1 ASC, col3_16, col3_11 ASC, col3_10 ASC, col3_7 ASC, col3_15), UNIQUE (col3_5, col3_10 DESC, col3_11 DESC, col3_12, col3_16 ASC, col3_6, col3_13, col3_9, col3_2 DESC, col3_4, col3_7 ASC, col1_1 ASC) STORING (col3_1, col3_14, col3_15), INDEX (col3_4 DESC) STORING (col3_5, col3_6, col3_11, col3_16), UNIQUE (col1_1)) INTERLEAVE IN PARENT table1 (col1_1);
        CREATE TABLE table4 (col4_0 FLOAT8 NOT NULL, PRIMARY KEY (col4_0 ASC), UNIQUE (col4_0 ASC) WHERE table4.col4_0 > 1.0:::FLOAT8);
        ALTER TABLE table1 INJECT STATISTICS e'[{"columns": ["col1_1"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 4759312512178489491, "histo_buckets": [{"distinct_range": 0, "num_eq": 50000000, "num_range": 0, "upper_bound": "37f995cc-003a-4c43-a966-27c57bf4ad21"}, {"distinct_range": 0, "num_eq": 30000, "num_range": 8407603935673917320, "upper_bound": "876007e1-b790-43a4-b886-e3b6c3ab4384"}, {"distinct_range": 200, "num_eq": 6293600967551416073, "num_range": 200, "upper_bound": "b5d69aed-a913-497b-b75f-95f771081739"}, {"distinct_range": 0, "num_eq": 6000000, "num_range": 0, "upper_bound": "c8e0c71c-1fd7-40ee-b4ae-d3eac1b1a1f9"}, {"distinct_range": 1000, "num_eq": 6580058833319890414, "num_range": 1000, "upper_bound": "ffffffff-ffff-ffff-ffff-ffffffffffff"}], "histo_col_type": "UUID", "name": "__auto__", "null_count": 0, "row_count": 4859652128523857415}, {"columns": ["col1_6"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 933579394258490697, "histo_col_type": "", "name": "__auto__", "null_count": 1322489089140522791, "row_count": 4859652128523857415}, {"columns": ["col1_10"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1020902363467022931, "histo_col_type": "", "name": "__auto__", "null_count": 4672709255451100832, "row_count": 4859652128523857415}, {"columns": ["col1_12"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 4766932519601755880, "histo_col_type": "", "name": "__auto__", "null_count": 2051050663558819599, "row_count": 4859652128523857415}, {"columns": ["col1_13"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2038867096504900692, "histo_buckets": [{"distinct_range": 0, "num_eq": 6086472873545648395, "num_range": 0, "upper_bound": "\\u00022D>EDra"}, {"distinct_range": 0, "num_eq": 60000, "num_range": 50000, "upper_bound": "\\u000c\\u000b{$\\u001f"}, {"distinct_range": 3415112987472565000, "num_eq": 5757361166338452733, "num_range": 7339694190995659817, "upper_bound": "FPQ \\u0019-"}, {"distinct_range": 1060409387402195600, "num_eq": 60000000, "num_range": 1060409387402195617, "upper_bound": "W\\u001d\\u000ce}\x7f5"}, {"distinct_range": 70000, "num_eq": 7284440434669967326, "num_range": 70000, "upper_bound": "nH\\rg!s/"}], "histo_col_type": "STRING", "name": "__auto__", "null_count": 1909724216764678320, "row_count": 4859652128523857415}, {"columns": ["col1_15"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1964309619621378751, "histo_col_type": "", "name": "__auto__", "null_count": 681798553106057440, "row_count": 4859652128523857415}, {"columns": ["col1_3"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 4518247833308447269, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 4859652128523857415}, {"columns": ["col1_4"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 494516793883862678, "histo_col_type": "", "name": "__auto__", "null_count": 1096339858009045305, "row_count": 4859652128523857415}, {"columns": ["col1_5"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 542260620078708763, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 4859652128523857415}, {"columns": ["col1_7"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2255299417484607648, "histo_col_type": "FLOAT4", "name": "__auto__", "null_count": 0, "row_count": 4859652128523857415}, {"columns": ["col1_18"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2662303259165191324, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 4859652128523857415}, {"columns": ["col1_2"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 3427361976053948329, "histo_col_type": "", "name": "__auto__", "null_count": 4519633178733216212, "row_count": 4859652128523857415}, {"columns": ["col1_14"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1786122615651910406, "histo_col_type": "", "name": "__auto__", "null_count": 1213707465247936035, "row_count": 4859652128523857415}, {"columns": ["col1_0"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1905022129703775650, "histo_col_type": "", "name": "__auto__", "null_count": 520117440895205356, "row_count": 4859652128523857415}, {"columns": ["col1_8"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 301829841877015393, "histo_col_type": "", "name": "__auto__", "null_count": 1943420403681495546, "row_count": 4859652128523857415}, {"columns": ["col1_9"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2636687812314286905, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 4859652128523857415}, {"columns": ["col1_11"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 3009584876558089447, "histo_col_type": "", "name": "__auto__", "null_count": 2358148140924975948, "row_count": 4859652128523857415}, {"columns": ["col1_16"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 248084219587940301, "histo_buckets": [{"distinct_range": 0, "num_eq": 300, "num_range": 0, "upper_bound": "1\\u001b"}, {"distinct_range": 4334.689050771067, "num_eq": 9168173067799604738, "num_range": 9000, "upper_bound": "1] \\u0016\\u0016U>k"}, {"distinct_range": 3542954137821616600, "num_eq": 702480323427157046, "num_range": 4194258620904654350, "upper_bound": "F{v8aX{0\\u001e"}, {"distinct_range": 1483613237.9944403, "num_eq": 300000, "num_range": 7000000000, "upper_bound": "G\\u0015\\u0013nv/"}, {"distinct_range": 82.94256592063772, "num_eq": 5000000000, "num_range": 90, "upper_bound": "I\\u0018j~?Xc\\u0008"}, {"distinct_range": 966593751844801700, "num_eq": 0, "num_range": 1594762163485470698, "upper_bound": "dWMo\\u0006\\u0007"}, {"distinct_range": 0, "num_eq": 50000, "num_range": 100000000000, "upper_bound": "dlm"}, {"distinct_range": 5000, "num_eq": 5587567521100962150, "num_range": 5000, "upper_bound": "fO"}, {"distinct_range": 900000, "num_eq": 2195409342129422392, "num_range": 900000, "upper_bound": "ik\\nf\\u0005\\u000b"}], "histo_col_type": "STRING", "name": "__auto__", "null_count": 0, "row_count": 4859652128523857415}, {"columns": ["col1_17"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1035252065349266563, "histo_col_type": "", "name": "__auto__", "null_count": 4109540553017965520, "row_count": 4859652128523857415}]':::JSONB;
        ALTER TABLE table2 INJECT STATISTICS '[{"columns": ["col2_12"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2352408685425312678, "histo_buckets": [{"distinct_range": 0, "num_eq": 3804678859770883348, "num_range": 0, "upper_bound": "001101101010010111110110"}, {"distinct_range": 6060740755.9600525, "num_eq": 8000, "num_range": 7000000000, "upper_bound": "011011111001010100111111"}, {"distinct_range": 0, "num_eq": 6210888757463293952, "num_range": 300000000, "upper_bound": "011111111111111111111111"}, {"distinct_range": 47667.482669291334, "num_eq": 5000000000, "num_range": 70000, "upper_bound": "100001001011101000110000"}, {"distinct_range": 352596629.65340817, "num_eq": 7032538812671661978, "num_range": 500000000, "upper_bound": "110110111000011000010011"}], "histo_col_type": "BIT(24)", "name": "__auto__", "null_count": 2157941634394962754, "row_count": 2670036992745511194}, {"columns": ["col2_13"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2058500975616824480, "histo_buckets": [{"distinct_range": 0, "num_eq": 0, "num_range": 0, "upper_bound": "-1.5532141596743272"}, {"distinct_range": 5658479740676143000, "num_eq": 60000000000, "num_range": 5658479740676142693, "upper_bound": "-0.9909751202384744"}, {"distinct_range": 10, "num_eq": 60000000000, "num_range": 10, "upper_bound": "5e-324"}, {"distinct_range": 20, "num_eq": 2785755476394450138, "num_range": 20, "upper_bound": "0.08006571544946262"}, {"distinct_range": 1100605493913549200, "num_eq": 30000000, "num_range": 4097351824982046584, "upper_bound": "0.0946474123885751"}, {"distinct_range": 0, "num_eq": 2343873084570065439, "num_range": 400, "upper_bound": "0.10103346051829748"}, {"distinct_range": 928528949565824500, "num_eq": 1204575637817092839, "num_range": 2169643035543797289, "upper_bound": "1.1070019623730833"}, {"distinct_range": 400000, "num_eq": 500000, "num_range": 400000, "upper_bound": "1.1874974863869578"}, {"distinct_range": 7000897.748931568, "num_eq": 800, "num_range": 50000000, "upper_bound": "3.4028234663852886e+38"}], "histo_col_type": "FLOAT8", "name": "__auto__", "null_count": 9782557728041518, "row_count": 2670036992745511194}, {"columns": ["col2_16"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1931878395189970139, "histo_col_type": "", "name": "__auto__", "null_count": 2165367322484962501, "row_count": 2670036992745511194}, {"columns": ["col1_1"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 802789369203980513, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 2670036992745511194}, {"columns": ["col2_5"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 307696541063475070, "histo_col_type": "", "name": "__auto__", "null_count": 1652162886725071704, "row_count": 2670036992745511194}, {"columns": ["col2_8"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2293074607246460489, "histo_col_type": "", "name": "__auto__", "null_count": 1195437113003786167, "row_count": 2670036992745511194}, {"columns": ["col2_10"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2022827613601346749, "histo_col_type": "", "name": "__auto__", "null_count": 973819937333064244, "row_count": 2670036992745511194}, {"columns": ["col2_11"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2546855125814635582, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 2670036992745511194}, {"columns": ["col2_15"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 644112382362372763, "histo_buckets": [{"distinct_range": 0, "num_eq": 5000000000, "num_range": 0, "upper_bound": "-1886724703"}, {"distinct_range": 600, "num_eq": 0, "num_range": 600, "upper_bound": "-1581517028"}, {"distinct_range": 0, "num_eq": 400, "num_range": 7000000, "upper_bound": "-1112729193"}, {"distinct_range": 2507456685067545000, "num_eq": 1191477352894920587, "num_range": 5575209544900073304, "upper_bound": "-877996767"}, {"distinct_range": 0, "num_eq": 60, "num_range": 8531456317539011725, "upper_bound": "262061866"}, {"distinct_range": 0, "num_eq": 6000000000, "num_range": 2253658199271189760, "upper_bound": "943644461"}, {"distinct_range": 0, "num_eq": 1000, "num_range": 0, "upper_bound": "1753244125"}], "histo_col_type": "INT4", "name": "__auto__", "null_count": 0, "row_count": 2670036992745511194}, {"columns": ["col2_2"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 200951358231367394, "histo_col_type": "", "name": "__auto__", "null_count": 71924787387219421, "row_count": 2670036992745511194}, {"columns": ["col2_4"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2471493557765313804, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 2670036992745511194}, {"columns": ["col2_7"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1968494124251134252, "histo_col_type": "", "name": "__auto__", "null_count": 2494997710593171373, "row_count": 2670036992745511194}, {"columns": ["col2_9"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 66417215879683561, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 2670036992745511194}, {"columns": ["col2_14"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1903623774981900972, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 2670036992745511194}, {"columns": ["col2_17"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 932275164024996032, "histo_buckets": [{"distinct_range": 0, "num_eq": 30000000, "num_range": 0, "upper_bound": "1976-03-05 18:42:34.000899+00:00"}, {"distinct_range": 49246745218.86555, "num_eq": 1000000000, "num_range": 100000000000, "upper_bound": "1985-05-10 12:21:10.000411+00:00"}, {"distinct_range": 1144804828792499300, "num_eq": 500000, "num_range": 1604339886825391793, "upper_bound": "1988-06-27 08:46:15.000282+00:00"}, {"distinct_range": 0, "num_eq": 0, "num_range": 5000, "upper_bound": "1990-03-09 18:00:32.000235+00:00"}, {"distinct_range": 0, "num_eq": 900000, "num_range": 50, "upper_bound": "1992-01-14 04:37:34.000354+00:00"}], "histo_col_type": "TIMESTAMPTZ", "name": "__auto__", "null_count": 2294034167196506479, "row_count": 2670036992745511194}, {"columns": ["col2_1"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 360313007192583723, "histo_buckets": [{"distinct_range": 0, "num_eq": 5799737957412212252, "num_range": 0, "upper_bound": "360412220"}, {"distinct_range": 400, "num_eq": 7484000236409056070, "num_range": 400, "upper_bound": "557128193"}, {"distinct_range": 85568.8868910887, "num_eq": 8000, "num_range": 600000, "upper_bound": "875212605"}, {"distinct_range": 0, "num_eq": 4348020482019873766, "num_range": 8415944035646047673, "upper_bound": "2466308165"}, {"distinct_range": 0, "num_eq": 700000000, "num_range": 1500950644955567238, "upper_bound": "3906406924"}, {"distinct_range": 6000, "num_eq": 9000000, "num_range": 6000, "upper_bound": "4022266954"}, {"distinct_range": 800000, "num_eq": 8356036722105780566, "num_range": 800000, "upper_bound": "4076783971"}, {"distinct_range": 0, "num_eq": 600000, "num_range": 0, "upper_bound": "4206156922"}], "histo_col_type": "REGTYPE", "name": "__auto__", "null_count": 0, "row_count": 2670036992745511194}, {"columns": ["col2_3"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1059301006304944543, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 2670036992745511194}, {"columns": ["col2_6"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 98217685918795789, "histo_col_type": "", "name": "__auto__", "null_count": 304750429851359421, "row_count": 2670036992745511194}]':::JSONB;
        ALTER TABLE table3 INJECT STATISTICS '[{"columns": ["col3_2"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1172779417514773602, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 8791980037187484719}, {"columns": ["col3_4"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 3801582508299596799, "histo_buckets": [{"distinct_range": 0, "num_eq": 2042845016020005181, "num_range": 0, "upper_bound": "BOX(-1.4673160798284324 0.08093608710719909,0.45694098427325525 0.37489875217914065)"}, {"distinct_range": 0, "num_eq": 5112234821019019230, "num_range": 5197412745161489932, "upper_bound": "BOX(-0.4133808070868671 -0.7923920999105879,-0.3675585799175429 0.40256057726147)"}, {"distinct_range": 4388649825307045400, "num_eq": 100485548948101008, "num_range": 8588213920206349651, "upper_bound": "BOX(-0.2474666130435308 0.048274199882943014,2.4728347752517017 0.8121328452377921)"}, {"distinct_range": 7935394413698584000, "num_eq": 561686151880300134, "num_range": 7935394413698583793, "upper_bound": "BOX(0.5294644107746166 -0.2151441064040801,0.8074268806225673 -0.16105255207783653)"}], "histo_col_type": "BOX2D", "name": "__auto__", "null_count": 3943376705898446748, "row_count": 8791980037187484719}, {"columns": ["col3_6"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 8284246454371634751, "histo_col_type": "", "name": "__auto__", "null_count": 5134935566388386533, "row_count": 8791980037187484719}, {"columns": ["col3_11"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 245668713830205396, "histo_col_type": "", "name": "__auto__", "null_count": 4435522962386382657, "row_count": 8791980037187484719}, {"columns": ["col3_14"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 3191506400454421248, "histo_col_type": "", "name": "__auto__", "null_count": 2351729551953261733, "row_count": 8791980037187484719}, {"columns": ["col3_15"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 3780704692037946496, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 8791980037187484719}, {"columns": ["col3_1"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 6498640135231981037, "histo_col_type": "", "name": "__auto__", "null_count": 200329113444812405, "row_count": 8791980037187484719}, {"columns": ["col3_5"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 3602990677230617190, "histo_buckets": [{"distinct_range": 0, "num_eq": 400, "num_range": 0, "upper_bound": "-32768"}, {"distinct_range": 135203635012131070, "num_eq": 1263038999356343727, "num_range": 3582644952817341768, "upper_bound": "1"}, {"distinct_range": 298400101330536770, "num_eq": 0, "num_range": 298400101330536742, "upper_bound": "2748031237690416454"}, {"distinct_range": 0, "num_eq": 500000000, "num_range": 1000000000, "upper_bound": "5567842697980837559"}], "histo_col_type": "INT8", "name": "__auto__", "null_count": 3901828993323396501, "row_count": 8791980037187484719}, {"columns": ["col3_10"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 8782398184490159777, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 8791980037187484719}, {"columns": ["col3_16"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 7473184748467020548, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 8791980037187484719}, {"columns": ["col1_1"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 7723050176247151643, "histo_buckets": [{"distinct_range": 0, "num_eq": 6000000, "num_range": 0, "upper_bound": "00000000-0000-0000-0000-000000000000"}, {"distinct_range": 31.385197649760478, "num_eq": 90000000, "num_range": 60, "upper_bound": "936abe96-8479-4b03-8e20-af2a25b27268"}, {"distinct_range": 391.49043000016485, "num_eq": 2511048207208629790, "num_range": 500, "upper_bound": "ebceddf4-7b79-4367-8665-42d67b1533fa"}], "histo_col_type": "UUID", "name": "__auto__", "null_count": 0, "row_count": 8791980037187484719}, {"columns": ["col3_7"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 5304184753281240785, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 8791980037187484719}, {"columns": ["col3_8"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 351885508692809805, "histo_col_type": "", "name": "__auto__", "null_count": 3717409340418438146, "row_count": 8791980037187484719}, {"columns": ["col3_9"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 4985673885925322097, "histo_col_type": "", "name": "__auto__", "null_count": 887308669837667434, "row_count": 8791980037187484719}, {"columns": ["col3_3"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 6555896033044150325, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 8791980037187484719}, {"columns": ["col3_13"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2148864888912303475, "histo_col_type": "", "name": "__auto__", "null_count": 3293086849071163743, "row_count": 8791980037187484719}, {"columns": ["col3_12"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 6423968870731481977, "histo_col_type": "", "name": "__auto__", "null_count": 4789239026896080561, "row_count": 8791980037187484719}]':::JSONB;
        ALTER TABLE table4 INJECT STATISTICS '[{"columns": ["col4_0"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 319279183, "histo_buckets": [{"distinct_range": 0, "num_eq": 2899987218713901584, "num_range": 0, "upper_bound": "-0.32833732947651156"}, {"distinct_range": 1056.5413413153858, "num_eq": 100000000, "num_range": 4000, "upper_bound": "-0.1891214778527421"}, {"distinct_range": 5967580707159511000, "num_eq": 40000000000, "num_range": 5967580707159510614, "upper_bound": "0.0"}, {"distinct_range": 0, "num_eq": 300000000, "num_range": 3074113934131121859, "upper_bound": "0.08324711521433614"}, {"distinct_range": 900000, "num_eq": 400000000, "num_range": 900000, "upper_bound": "0.5727766604112363"}], "histo_col_type": "FLOAT8", "name": "__auto__", "null_count": 0, "row_count": 700000000}]':::JSONB;
        ALTER TABLE table2 ADD FOREIGN KEY (col1_1) REFERENCES table3 (col1_1) ON UPDATE CASCADE;
        ALTER TABLE table2 ADD FOREIGN KEY (col2_5) REFERENCES table1 (col1_16) ON UPDATE RESTRICT;
        CREATE TYPE rand_typ_0 AS ENUM ('o', 'qpelg', 'xc', 'l');
        CREATE TYPE rand_typ_1 AS ENUM ('okt', 'vokfzh', 'z');
        CREATE TYPE rand_typ_2 AS ENUM ('ndsh', 'hy');
        CREATE TYPE rand_typ_3 AS ENUM ('h');
        CREATE TYPE rand_typ_4 AS ENUM ('oar', 'xj', 'nqtyu');
        ;
    rsg_test.go:577: 
        SET CLUSTER SETTING sql.defaults.drop_enum_value.enabled = true;
        SET enable_drop_enum_value = true;
        
        CREATE TYPE greeting AS ENUM ('hello', 'howdy', 'hi', 'good day', 'morning');
        CREATE TABLE IF NOT EXISTS seed AS
        	SELECT
        		g::INT2 AS _int2,
        		g::INT4 AS _int4,
        		g::INT8 AS _int8,
        		g::FLOAT4 AS _float4,
        		g::FLOAT8 AS _float8,
        		'2001-01-01'::DATE + g AS _date,
        		'2001-01-01'::TIMESTAMP + g * '1 day'::INTERVAL AS _timestamp,
        		'2001-01-01'::TIMESTAMPTZ + g * '1 day'::INTERVAL AS _timestamptz,
        		g * '1 day'::INTERVAL AS _interval,
        		g % 2 = 1 AS _bool,
        		g::DECIMAL AS _decimal,
        		g::STRING AS _string,
        		g::STRING::BYTES AS _bytes,
        		substring('00000000-0000-0000-0000-' || g::STRING || '00000000000', 1, 36)::UUID AS _uuid,
        		'0.0.0.0'::INET + g AS _inet,
        		g::STRING::JSONB AS _jsonb,
        		enum_range('hello'::greeting)[g] as _enum
        	FROM
        		generate_series(1, 5) AS g;
        
        INSERT INTO seed DEFAULT VALUES;
        CREATE INDEX on seed (_int8, _float8, _date);
        CREATE INVERTED INDEX on seed (_jsonb);
        ;
    rsg_test.go:579: 
    rsg_test.go:580: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestRandomSyntaxSQLSmith109561685
--- FAIL: TestRandomSyntaxSQLSmith (301.06s)
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
make stressrace TESTS=TestRandomSyntaxSQLSmith PKG=./pkg/sql/tests TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #64592 sql/tests: TestRandomSyntaxSQLSmith failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<details><summary>Internal log</summary>
<p>

```
mjibson marked as alumn{us/a}; resolving to rafiss instead
```
</p>
</details>

/cc @cockroachdb/sql-queries @rafiss
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxSQLSmith.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/colfetcher/cfetcher.go
+++ b/pkg/sql/colfetcher/cfetcher.go
@@ -437,10 +437,14 @@ func (rf *cFetcher) Init(
 	var err error
 
 	var neededCols util.FastIntSet
-	// Scan through the entire columns map to see which columns are
-	// required.
-	if numNeededCols := tableArgs.ValNeededForCol.Len(); cap(table.neededColsList) < numNeededCols {
+	numNeededCols := tableArgs.ValNeededForCol.Len()
+	// Scan through the entire columns map to see which columns are required and
+	// which columns are not needed (the latter will be set to all null values).
+	if cap(table.neededColsList) < numNeededCols {
 		table.neededColsList = make([]int, 0, numNeededCols)
+	}
+	if cap(table.notNeededColOrdinals) < len(colDescriptors)-numNeededCols {
+		table.notNeededColOrdinals = make([]int, 0, len(colDescriptors)-numNeededCols)
 	}
 	for i := range colDescriptors {
 		//gcassert:bce
@@ -458,20 +462,12 @@ func (rf *cFetcher) Init(
 			case descpb.SystemColumnKind_TABLEOID:
 				table.oidOutputIdx = idx
 			}
+		} else {
+			table.notNeededColOrdinals = append(table.notNeededColOrdinals, idx)
 		}
 	}
 	sort.Ints(table.neededColsList)
-
-	// Find the set of columns for which vectors will **not** be properly
-	// populated.
-	if numNeededCols := tableArgs.ValNeededForCol.Len(); cap(table.notNeededColOrdinals) < len(rf.typs)-numNeededCols {
-		table.notNeededColOrdinals = make([]int, 0, len(rf.typs)-numNeededCols)
-	}
-	for i := 0; i < len(rf.typs); i++ {
-		if !tableArgs.ValNeededForCol.Contains(i) {
-			table.notNeededColOrdinals = append(table.notNeededColOrdinals, i)
-		}
-	}
+	sort.Ints(table.notNeededColOrdinals)
 
 	table.knownPrefixLength = len(rowenc.MakeIndexKeyPrefix(codec, table.desc, table.index.GetID()))
 
@@ -1114,14 +1110,12 @@ func (rf *cFetcher) nextBatch(ctx context.Context) (coldata.Batch, error) {
 			if emitBatch {
 				rf.pushState(stateResetBatch)
 				rf.finalizeBatch()
-				rf.machine.rowIdx = 0
 				return rf.machine.batch, nil
 			}
 
 		case stateEmitLastBatch:
 			rf.machine.state[0] = stateFinished
-			rf.machine.batch.SetLength(rf.machine.rowIdx)
-			rf.machine.rowIdx = 0
+			rf.finalizeBatch()
 			return rf.machine.batch, nil
 
 		case stateFinished:
@@ -1500,9 +1494,10 @@ func (rf *cFetcher) finalizeBatch() {
 	// batch is materialized (i.e. values are converted to datums), the
 	// conversion of unset values might encounter an error.
 	for _, notNeededIdx := range rf.table.notNeededColOrdinals {
-		rf.machine.batch.ColVec(notNeededIdx).Nulls().SetNulls()
+		rf.machine.colvecs[notNeededIdx].Nulls().SetNulls()
 	}
 	rf.machine.batch.SetLength(rf.machine.rowIdx)
+	rf.machine.rowIdx = 0
 }
 
 // getCurrentColumnFamilyID returns the column family id of the key in

--- a/pkg/sql/logictest/testdata/logic_test/vectorize_types
+++ b/pkg/sql/logictest/testdata/logic_test/vectorize_types
@@ -118,3 +118,21 @@ query T
 SELECT 'a'::"char" FROM t47131_0
 ----
 a
+
+# Regression test for the cFetcher not setting the not-needed columns to all
+# nulls in some cases (#64676).
+statement ok
+CREATE TABLE t64676 (
+  i INT,
+  d DATE,
+  u UUID
+);
+INSERT INTO t64676 VALUES
+  (1, '2021-05-05', '00000000-0000-0000-0000-100000000000'),
+  (2, '2021-05-05', '00000000-0000-0000-0000-200000000000'),
+  (3, '2021-05-05', '00000000-0000-0000-0000-300000000000'),
+  (4, '2021-05-05', '00000000-0000-0000-0000-400000000000'),
+  (5, '2021-05-05', '00000000-0000-0000-0000-500000000000')
+
+statement ok
+SELECT i + d FROM t64676

